app/src/main/java/com/arctouch/codechallenge/home/HomeActivity.kt

```diff
@@ -2,29 +2,60 @@ package com.arctouch.codechallenge.home
 
 import android.os.Bundle
 import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.arctouch.codechallenge.R
-import com.arctouch.codechallenge.api.TmdbApi
-import com.arctouch.codechallenge.base.BaseActivity
-import com.arctouch.codechallenge.data.Cache
-import io.reactivex.android.schedulers.AndroidSchedulers
-import io.reactivex.schedulers.Schedulers
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.PaginationScrollListener
 import kotlinx.android.synthetic.main.home_activity.*
+import androidx.lifecycle.ViewModelProviders
+
+
+class HomeActivity : AppCompatActivity() {
+
+    private var adapter = HomeAdapter()
+    lateinit var viewModel : HomeViewModel
 
-class HomeActivity : BaseActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.home_activity)
+        viewModel = ViewModelProviders.of(this).get(HomeViewModel::class.java)
+        configRecyclerView()
+        viewModel.onMoviesResult.observe(this, Observer<MutableList<Movie>> { this.onMoviesResult(it) })
+        viewModel.startService()
+    }
+
+    private fun onMoviesResult(it: MutableList<Movie>) {
+        if (progressBar.visibility == View.VISIBLE) {
+            progressBar.visibility = View.GONE
+        } else {
+            adapter.removeLoadingFooter()
+        }
+        adapter.addMovies(it)
+        if (!viewModel.isLastPage) {
+            adapter.addLoadingFooter()
+        }
+    }
+
+    private fun configRecyclerView() {
+        val linearLayoutManager = LinearLayoutManager(this)
+        recyclerView.adapter = adapter
+        recyclerView.layoutManager = linearLayoutManager
+        recyclerView.addOnScrollListener(object : PaginationScrollListener(linearLayoutManager) {
+            override fun loadMoreItems() {
+                viewModel.isLoading = true
+                viewModel.loadPage()
+            }
+
+            override fun isLastPage(): Boolean {
```

Comment:
A more concise way to write one-line code in Kotlin is:

override fun isLastPage() = viewModel.isLastPage

Reply:
ok, great!